### PR TITLE
Fixes 1026:Implemented State Management For Images in Infobox

### LINKED ISSUE #1026: Implement Images in Infobox using current State Management Pattern in Project.
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Expected behavior:**
Implement Images in Infobox using state management. It will be very useful when refactoring, or implementing features that would otherwise require many changes across the whole codebase, or touching many components. It will also increase code coverage.

### PR DESCRIPTION
**Pull request by praveenojha33**
Fixes #1026 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:
Implemented State Management For Images in Infobox
Surge Link: https://pr-1027-fossasia-susper.surge.sh

### CODE REVIEW

File: src/app/actions/knowledge.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+  CONTENT_CHANGE: type('[Knowledge] Content Change'),
+  IMAGE_CHANGE: type('[Knowledge] Image Change')
 };
-export class SearchAction implements Action {
-  type = ActionTypes.CHANGE;
+export class SearchContent implements Action {
+  type = ActionTypes.CONTENT_CHANGE;
+  constructor(public payload: any) {}
+}
+export class SearchImage implements Action {
```
Comment (simsausaurabh) [inline src/app/actions/knowledge.ts]: Make it ```SearchImageAction```

File: src/app/actions/knowledge.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
 import { Action } from '@ngrx/store';
 import { type } from '../utils';
 export const ActionTypes = {
-  CHANGE: type('[Knowledge] Change')
+  CONTENT_CHANGE: type('[Knowledge] Content Change'),
+  IMAGE_CHANGE: type('[Knowledge] Image Change')
 };
-export class SearchAction implements Action {
-  type = ActionTypes.CHANGE;
+export class SearchContent implements Action {
```
Comment (simsausaurabh) [inline src/app/actions/knowledge.ts]: Make it ```SearchContentAction```

File: src/app/infobox/infobox.component.ts [hunk 3]
```diff
@@ -32,20 +31,14 @@ export class InfoboxComponent implements OnInit {
...
-            this.image = res.RelatedTopics[0].Icon.URL;
-          } else {
-            this.image = '';
-          }
-          }
-
-        );
+    this.image_response$ = store.select(fromRoot.getKnowledgeImage);
+    this.image_response$.subscribe(res => {
+      if (typeof res.RelatedTopics !== "undefined") {
```
Comment (simsausaurabh) [inline src/app/infobox/infobox.component.ts]: Use single quotes in undefined.

File: src/app/reducers/knowledge.ts [hunk 4]
```diff
@@ -3,27 +3,38 @@ import { combineReducers } from '@ngrx/store';
...
+        image_response: image_response
       });
     }
     default: {
       return state;
     }
   }
 }
-export const getresponse = (state: State) => state.response;
+export const getcontentresponse = (state: State) => state.content_response;
```
Comment (simsausaurabh) [inline src/app/reducers/knowledge.ts]: Always use camel case for methods.

File: src/app/reducers/knowledge.ts [hunk 5]
```diff
@@ -3,27 +3,38 @@ import { combineReducers } from '@ngrx/store';
...
+export const getimageresponse = (state: State) => state.image_response;
```
Comment (simsausaurabh) [inline src/app/reducers/knowledge.ts]: Always use camel case for methods.

File: src/app/effects/knowledge.ts [hunk 6]
```diff
@@ -45,15 +45,27 @@ export class KnowledgeEffects {
...
                     return empty();
             }
 
 
         });
+        this.knowledgeservice.getImage(querypay.query)
+        .takeUntil(nextSearch$)
+        .subscribe((response) => {
+          if (response.RelatedTopics.length > 0) {
+            this.store.dispatch(new knowledge.SearchImage(response));
```
Comment (simsausaurabh) [inline src/app/effects/knowledge.ts]: What is the type of response here? If it remains fixed, then make a change from ```any``` to the response type in action payload in the action defined above.

**Comment by praveenojha33** [comment]
@simsausaurabh All the changes have been made please review again.

**Pull request merged**